zaino-serve/src/rpc/service.rs

```
@@ -1629,46 +1793,65 @@ impl CompactTxStreamer for GrpcClient {
+                                    .send(Err(tonic::Status::unknown(
+                                        "Error: Satoshis out of range. Failed to convert to i64.",
+                                    )))
+                                    .await;
+                                return;
+                            }
+                        };
+                        let utxo_reply = GetAddressUtxosReply {
+                            address: utxo.address.to_string(),
+                            txid: utxo.txid.0.to_vec(),
+                            index: checked_index,
+                            script: utxo.script.as_ref().to_vec(),
+                            value_zat: checked_satoshis,
+                            height: utxo.height.0 as u64,
+                        };
+                        if channel_tx.send(Ok(utxo_reply)).await.is_err() {
                             return;
                         }
-                    };
-                    let utxo_reply = GetAddressUtxosReply {
-                        address: utxo.address.to_string(),
-                        txid: utxo.txid.0.to_vec(),
-                        index: checked_index,
-                        script: utxo.script.as_ref().to_vec(),
-                        value_zat: checked_satoshis,
-                        height: utxo.height.0 as u64,
-                    };
-                    if channel_tx.send(Ok(utxo_reply)).await.is_err() {
+                    }
+                })
+                .await;
+                match timeout {
+                    Ok(_) => {
+                        return;
+                    }
+                    Err(_) => {
+                        channel_tx
+                            .send(Err(tonic::Status::deadline_exceeded(
+                                "Error: get_mempool_stream gRPC request timed out",
+                            )))
+                            .await
+                            .ok();
```

here `.ok()` is used in a similar way to `let _ =`, is this because in all these calls we dont care if it errors? My opinion is we will always want to handle errors with either an error return or expect (if it cant fail). if an error is returned then the handler of this thread can decide to ignore the Err case